wdl-engine/src/backend/crankshaft.rs

```diff
@@ -318,23 +357,68 @@ impl TaskExecutionBackend for CrankshaftBackend {
         })
     }
 
-    fn container_root_dir(&self) -> Option<&Path> {
-        Some(Path::new("/mnt/task"))
+    fn guest_work_dir(&self) -> Option<&Path> {
+        Some(Path::new(GUEST_WORK_DIR))
+    }
+
+    fn localize_inputs<'a, 'b, 'c, 'd>(
+        &'a self,
+        downloader: &'b dyn Downloader,
+        inputs: &'c mut [crate::eval::Input],
+    ) -> BoxFuture<'d, Result<()>>
+    where
+        'a: 'd,
+        'b: 'd,
+        'c: 'd,
+        Self: 'd,
+    {
+        async {
+            // Construct a trie for mapping input guest paths
+            let mut trie = InputTrie::default();
+            for input in inputs.iter() {
+                trie.insert(input)?;
+            }
+
+            for (index, guest_path) in trie.calculate_guest_paths(GUEST_INPUTS_DIR)? {
+                inputs[index].set_guest_path(guest_path);
+            }
+
+            // Localize all inputs
+            // TODO: only do this for local task execution
+            for input in inputs {
+                // TODO: parallelize the downloads
```

Comment:
Should we just go ahead and knock this out in this PR? That seems relatively straightforward with a `FuturesUnordered`.

Reply:
It requires a little more planning as we'll probably want to limit at most X concurrent downloads (i.e. a semaphore in the downloader) with a configuration option to control the limit.

I'll open an issue to implement it.

Reply:
Okay, if you think it's too much to fit in here, that's fine.

Reply:
Opened https://github.com/stjude-rust-labs/wdl/issues/388.